File: server/server-core/src/main/java/com/simple/castle/server/physic/unit/GroundObject.java
package com.simple.castle.server.physic.unit;

import com.badlogic.gdx.math.Vector3;
import com.badlogic.gdx.physics.bullet.dynamics.btRigidBody;

public class GroundObject extends KinematicObject {

    private final Vector3 position = new Vector3();

    public GroundObject(btRigidBody.btRigidBodyConstructionInfo constructionInfo) {
        super(constructionInfo);
    }

    public Vector3 getPosition() {
        return body.getWorldTransform().getTranslation(position);
    }
}

File: server/server-core/src/main/java/com/simple/castle/server/kt/physic/GroundObject.kt
package com.simple.castle.server.kt.physic

import com.badlogic.gdx.math.Vector3
import com.badlogic.gdx.physics.bullet.dynamics.btRigidBody

class GroundObject(constructionInfo: btRigidBody.btRigidBodyConstructionInfo) : KinematicObject(constructionInfo) {
    val position = Vector3()
        get() = body.worldTransform.getTranslation(field)
}